inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp

```diff
@@ -0,0 +1,1943 @@
+        blk_size = 8;
+    }
+}
+
+void MKLDNNReduceNode::execute(mkldnn::stream strm) {
+    auto &dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+    auto &srcMemPtr = getParentEdgeAt(REDUCE_DATA)->getMemoryPtr();
+    auto &srcIndexesMemPtr = getParentEdgeAt(REDUCE_INDEXES)->getMemoryPtr();
+
+    const auto idx_data = reinterpret_cast<const int32_t *>(srcIndexesMemPtr->GetData());
+    size_t dst_size = dstMemPtr->GetSize();
+    src_dims = getParentEdgeAt(REDUCE_DATA)->getDesc().getDims();
+    src_strides = getParentEdgeAt(REDUCE_DATA)->getDesc().getBlockingDesc().getStrides();
+    dims_size = src_dims.size();
+    calc_process_dst_dims(idx_data);
+
+    if (dims_size <= 5) {
+        if (dims_size == 5) {
+            SET_SRC_DIM_VALUE(src_dims[0], src_dims[1], src_dims[2], src_dims[3], src_dims[4]);
+            SET_DST_DIM_VALUE(process_dst_dims[0], process_dst_dims[1], process_dst_dims[2], process_dst_dims[3], process_dst_dims[4]);
+        } else if (dims_size == 4) {
+            SET_SRC_DIM_VALUE(src_dims[0], src_dims[1], 1, src_dims[2], src_dims[3]);
+            SET_DST_DIM_VALUE(process_dst_dims[0], process_dst_dims[1], 1, process_dst_dims[2], process_dst_dims[3]);
+        } else if (dims_size == 3) {
+            SET_SRC_DIM_VALUE(1, src_dims[0], 1, src_dims[1], src_dims[2]);
+            SET_DST_DIM_VALUE(1, process_dst_dims[0], 1, process_dst_dims[1], process_dst_dims[2]);
+        } else if (dims_size == 2) {
+            SET_SRC_DIM_VALUE(src_dims[0], src_dims[1], 1, 1, 1);
+            SET_DST_DIM_VALUE(process_dst_dims[0], process_dst_dims[1], 1, 1, 1);
+        } else {
+            SET_SRC_DIM_VALUE(1, src_dims[0], 1, 1, 1);
+            SET_DST_DIM_VALUE(1, process_dst_dims[0], 1, 1, 1);
+        }
+
+        ReduceN = IB != OB && OB == 1;
+        ReduceC = IC != OC && OC == 1;
+        ReduceD = ID != OD && OD == 1;
+        ReduceH = IH != OH && OH == 1;
+        ReduceW = IW != OW && OW == 1;
+    }
+
+    const uint8_t *src_data = reinterpret_cast<const uint8_t *>(srcMemPtr->GetData()) +
+                   srcMemPtr->GetDescriptor().data.layout_desc.blocking.offset_padding *
+                   MKLDNNExtensionUtils::sizeOfDataType(mkldnn::memory::data_type(srcMemPtr->GetDescriptor().data.data_type));
+    uint8_t *dst_data = reinterpret_cast<uint8_t *>(dstMemPtr->GetData()) +
+                   dstMemPtr->GetDescriptor().data.layout_desc.blocking.offset_padding *
+                   MKLDNNExtensionUtils::sizeOfDataType(mkldnn::memory::data_type(dstMemPtr->GetDescriptor().data.data_type));
+    if (reduce_kernel && dims_size <= 5) {
+        reduce_type(src_data, dst_data, dst_size);
+    } else {
+        if (planar_layout) {
+            auto in_ptr = reinterpret_cast<const float *>(src_data);
+            auto out_ptr = reinterpret_cast<float *>(dst_data);
+            reduce_ref(in_ptr, out_ptr);
+        } else {
+            THROW_IE_EXCEPTION << "Only support plain layout on machine w/o sse42.";
```

Comment:
Please add information about layer type and name like in other exceptions

Reply:
Fixed. Thanks Dmitry!